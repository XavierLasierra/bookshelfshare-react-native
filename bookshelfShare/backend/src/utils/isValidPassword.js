const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
