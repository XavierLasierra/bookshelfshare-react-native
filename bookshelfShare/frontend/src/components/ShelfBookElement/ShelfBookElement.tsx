import React from 'react';
import {
  Text, TouchableOpacity, Image, View
} from 'react-native';
import { IShelfBookElement } from '../../types/interfaces';

import styles from './shelfBookElement.styles';

export default function ShelfBookElement({
  bookData, logo, navigation, location, shelfName
}: IShelfBookElement) {
  function handleBookDetailPage() {
    navigation.push('BookDetail',
      {
        bookData,
        logo
      });
  }

  return (
    <TouchableOpacity
      activeOpacity={0.8}
      onPress={handleBookDetailPage}
      testID="bookDetailButton"
      style={styles.container}
    >
      <Text style={styles.shelfLocation}>
        Row:
        {' '}
        {location[0] + 1}
        {' '}
        Column:
        {' '}
        {location[1] + 1}
        {' '}
        -
        {' '}
        {shelfName}
      </Text>
      <View style={{ flexDirection: 'row' }}>
        <Image
          style={styles.bookImage}
          source={{ uri: bookData?.images?.thumbnail || bookData?.images?.smallThumbnail }}
        />
        <View style={styles.informationContainer}>
          <View>
            <Text style={styles.bookTitle}>{bookData?.title}</Text>
            {bookData?.authors.map((author: string, index: number) => (index < 2
        && <Text key={`${author}-${bookData?.title}`}>{author}</Text>))}
            {bookData?.authors.length >= 3 && <Text>...</Text>}
            <Text>{bookData?.publisher}</Text>
          </View>
          <Text>
            ISBN13 -
            {' '}
            {bookData?.isbn?.ISBN13}
          </Text>
        </View>
      </View>
    </TouchableOpacity>
  );
}
