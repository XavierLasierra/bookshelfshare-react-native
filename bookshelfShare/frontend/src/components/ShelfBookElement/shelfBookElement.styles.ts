import {StyleSheet} from 'react-native';
import stylesConstants from '../../styles/styles.constants';

export default StyleSheet.create({
  container: {
    marginVertical: 10,
  },
  bookImage: {
    width: 100,
    height: 150,
    overflow: 'hidden',
    borderTopLeftRadius: 10,
    borderBottomLeftRadius: 10,
  },
  informationContainer: {
    backgroundColor: stylesConstants.colors.white,
    width: 210,
    height: 150,
    padding: stylesConstants.padding.small,
    borderTopRightRadius: 10,
    borderBottomRightRadius: 10,
    justifyContent: 'space-between',
  },
  bookTitle: {
    width: 175,
    fontWeight: 'bold',
    fontSize: stylesConstants.fontSize.mid,
  },
  bookAuthor: {
    fontSize: stylesConstants.fontSize.small,
  },
  bookPublisher: {
    fontSize: stylesConstants.fontSize.small,
    color: stylesConstants.colors.mainText,
  },
  bookIsbn: {
    fontSize: stylesConstants.fontSize.small,
    color: stylesConstants.colors.mainText,
  },
  shelfLocation: {
    fontWeight: 'bold',
    fontSize: stylesConstants.fontSize.mid,
  },
});
