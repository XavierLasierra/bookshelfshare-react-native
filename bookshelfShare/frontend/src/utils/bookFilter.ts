export function bookFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfListFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.bookData.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfLocationFilter(filterLocation: number[], books: any) {
  return books.filter(({ customInformation: { location } }: any) => JSON.stringify(location)
  === JSON.stringify(filterLocation));
}
