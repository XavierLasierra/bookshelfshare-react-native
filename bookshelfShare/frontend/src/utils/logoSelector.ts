import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
