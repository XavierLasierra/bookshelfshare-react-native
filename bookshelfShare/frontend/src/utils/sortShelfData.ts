export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
