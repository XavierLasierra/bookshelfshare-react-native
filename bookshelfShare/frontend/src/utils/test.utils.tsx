import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

function render(
  component: any,
  initialState: any
) {
  const store = configureStore(initialState);

  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    );
  }
  return rtlRender(component, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
