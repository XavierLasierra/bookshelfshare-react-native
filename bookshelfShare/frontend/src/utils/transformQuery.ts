interface Query {
    isbn?: string,
    inauthor?: string,
    intitle?: string,
    inpublisher?: string
}

export default function transformQuery(query: Query):string {
  const queryEntries = Object.entries(query);
  const url = queryEntries
    .reduce((acc, queryElement, index) => `${acc}${queryElement[0]}=${queryElement[1]}${index < queryEntries.length - 1 ? '+' : ''}`, '');
  return `${url}&key=${process.env.GOOGLE_API_KEY}`;
}
