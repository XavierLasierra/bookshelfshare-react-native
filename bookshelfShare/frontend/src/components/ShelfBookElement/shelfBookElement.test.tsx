import React from 'react';
import ShelfBookElement from './ShelfBookElement';
import { render, fireEvent } from '../../utils/test.utils';
import bookDetailsMock from '../../mocks/bookDetails.mock';

const navigation = {
  push: jest.fn()
};

describe('Given a ShelfBookElement component', () => {
  describe('When it is rendered', () => {
    describe('And bookData has tumbnail and less than 3 authors', () => {
      let screen: any;
      beforeEach(() => {
        screen = render(<ShelfBookElement
          navigation={navigation}
          bookData={bookDetailsMock[0]}
          location={[0, 0]}
          shelfName="My shelf"
        />);
      });
      test('Then should match the snapshot', () => {
        expect(screen).toMatchSnapshot();
      });

      describe('And you click on bookDetailButton', () => {
        test('Then navigation.push should have been called', () => {
          const bookDetailButton = screen.getByTestId('bookDetailButton');
          fireEvent.press(bookDetailButton);

          expect(navigation.push).toHaveBeenCalled();
        });
      });
    });
    describe('And bookData has no tumbnail and 3 authors', () => {
      test('Then should match the snapshot', () => {
        const screen = render(<ShelfBookElement
          navigation={navigation}
          bookData={bookDetailsMock[1]}
          location={[0, 0]}
          shelfName="My shelf"
        />);
        expect(screen).toMatchSnapshot();
      });
    });
  });
});
