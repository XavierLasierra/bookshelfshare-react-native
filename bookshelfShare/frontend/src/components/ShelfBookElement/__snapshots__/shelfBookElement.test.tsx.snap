// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ShelfBookElement component When it is rendered And bookData has no tumbnail and 3 authors Then should match the snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginVertical": 10,
      "opacity": 1,
    }
  }
  testID="bookDetailButton"
>
  <Text
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Row:
     
    1
     
    Column:
     
    1
     
    -
     
    My shelf
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "smallThumbnail2",
        }
      }
      style={
        Object {
          "borderBottomLeftRadius": 10,
          "borderTopLeftRadius": 10,
          "height": 150,
          "overflow": "hidden",
          "width": 100,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderBottomRightRadius": 10,
          "borderTopRightRadius": 10,
          "height": 150,
          "justifyContent": "space-between",
          "padding": 10,
          "width": 210,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "width": 175,
            }
          }
        >
          Title
        </Text>
        <Text>
          Orwell
        </Text>
        <Text>
          Another author
        </Text>
        <Text>
          ...
        </Text>
        <Text>
          Debolsillo
        </Text>
      </View>
      <Text>
        ISBN13 -
         
        4567765432123
      </Text>
    </View>
  </View>
</View>
`;

exports[`Given a ShelfBookElement component When it is rendered And bookData has tumbnail and less than 3 authors Then should match the snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginVertical": 10,
      "opacity": 1,
    }
  }
  testID="bookDetailButton"
>
  <Text
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Row:
     
    1
     
    Column:
     
    1
     
    -
     
    My shelf
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "thumbnail1",
        }
      }
      style={
        Object {
          "borderBottomLeftRadius": 10,
          "borderTopLeftRadius": 10,
          "height": 150,
          "overflow": "hidden",
          "width": 100,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderBottomRightRadius": 10,
          "borderTopRightRadius": 10,
          "height": 150,
          "justifyContent": "space-between",
          "padding": 10,
          "width": 210,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "width": 175,
            }
          }
        >
          1984
        </Text>
        <Text>
          George Orwell
        </Text>
        <Text>
          No publisher found
        </Text>
      </View>
      <Text>
        ISBN13 -
         
        9780451524935
      </Text>
    </View>
  </View>
</View>
`;
